src/mpi/romio/mpi-io/read.c

```diff
@@ -60,8 +60,6 @@ int MPI_File_read(MPI_File fh, void *buf, int count, MPI_Datatype datatype, MPI_
     return error_code;
 }
 
-/* prevent multiple definitions of this routine */
-#ifdef MPIO_BUILD_PROFILING
```

Comment:
On platforms that doesn't support weak symbols, we will build the code twice, once with every functions included but with top `MPI_xxx` functions defined as `PMPI_xxx`; and a second time with all the `MPI_xxx` functions but stripping all the internal function. Normally, both libraries are linked. When user use the profiling interface, they will replace the latter with their own profiling library.

If you remove this macro, the internal functions will be included in both libraries and causing duplication when linked together.

I am commenting based on how profiling works for core MPICH. ROMIO behaves similarly, right?

Reply:
Mac doesn't support weak symbols, so it can be used for testing.

Reply:
I tested this by adding '--disable-weak-symbols' at configure command line and without it.
Please let me know the case that failed this PR.

Reply:
Does MPICH CI include building on a Mac?

Reply:
Yes, we have macOS CI builds.